### PR TITLE
Feature/api personal

### PR DESCRIPTION
**Pull request by Aurora12**
Returns a list of objects, each containing
– channel: object (kind='', type != 'O' –> that is, personal chats)
– info: members count, unread count, etc.
– last_message: object (if any)
– last_user: object (if any, this is the author of last_message)

### CODE REVIEW

File: app/channel.go [hunk 1]
```diff
@@ -2491,3 +2491,11 @@ func (a *App) AssignCategory(channelId string, categoryId int32) (*model.Channel
 		return newChannel, nil
 	}
 }
+
+func (a *App) GetPersonalChannels(teamId string, userId string) (*model.ChannelSnapshotList, *model.AppError) {
```
Comment (ygworldr) [inline app/channel.go:2495]: Since it is a public method, I would add a comment even though the code is crystal clear! :grin:

File: app/team.go [hunk 2]
```diff
@@ -1622,3 +1622,7 @@ func (a *App) ClearTeamMembersCache(teamID string) {
 		page++
 	}
 }
+
+func (a *App) MainTeam() (*model.Team, *model.AppError) {
```
Comment (ygworldr) [inline app/team.go:1626]: A public function should have a comment… /pedentic.

File: store/store.go [hunk 3]
```diff
@@ -119,6 +119,8 @@ type TeamStore interface {
 
 	// GroupSyncedTeamCount returns the count of non-deleted group-constrained teams.
 	GroupSyncedTeamCount() (int64, *model.AppError)
+
+	MainTeam() (*model.Team, *model.AppError)
```
Comment (ygworldr) [inline store/store.go:123]: For consistency, I would add a comment here.

File: store/store.go [hunk 4]
```diff
@@ -210,6 +212,8 @@ type ChannelStore interface {
 
 	// GroupSyncedChannelCount returns the count of non-deleted group-constrained channels.
 	GroupSyncedChannelCount() (int64, *model.AppError)
+
```
Comment (ygworldr) [inline store/store.go:215]: For consistency, I would add a comment here.

File: app/channel.go (hunk 1 above)
Comment (Aurora12) [reply app/channel.go:2495]: :zap:

File: app/team.go (hunk 2 above)
Comment (Aurora12) [reply app/team.go:1626]: :zap:

File: store/store.go (hunk 3 above)
Comment (Aurora12) [reply store/store.go:123]: ⚡

File: store/store.go (hunk 4 above)
Comment (Aurora12) [reply store/store.go:215]: This is the original Mmost code, so I wouldn't comment on it until we work with it.

**Pull request merged**